### ISSUE TITLE
Error: Authenticate error: unauthorized_client

### ISSUE THREAD

**Issue by Kositch** [ssadams11/node-red-contrib-netatmo#28]
Since yesterday I receive this error (plugin version 0.3.0), plugin has not been yet updated to the new authentication method, so I suppose Netatmo stopped possibility to use plugins based on user/password authentication. Anyone got this error also? Is there any workaround? I suppose lots of people use Netatmo data in their automations in NR. Thanks!

**Comment by Multatultasaa** [comment]
I also had the same problem yesterday. I can't get weather information anymore with plugin 0.3.0 to NodeRed.

**Comment by S474N** [comment]
Same problem, but node gives the data from Netatmo server :)

**Comment by Kositch** [comment]
> Same problem, but node gives the data from Netatmo server :)

Sadly until first NR restart...

**Comment by S474N** [comment]
Restarted 5 days ago (have dockerized version).

**Comment by Kositch** [comment]
> Restarted 5 days ago (have dockerized version).

Have you tried to restart NR today? 5 days ago it worked even after the restart... But not after the restart yesterday.

**Comment by hflatoey** [comment]
Same problem here... Also tried to restart NR but that didn't help.

**Comment by Jayy77** [comment]
same problem - help/update very much appreciated!

**Comment by S474N** [comment]
You can try this plugin for Netatmo:
https://github.com/guidone/node-red-contrib-netatmo-dashboard

**Comment by Kositch** [comment]
> You can try this plugin for Netatmo:
> https://github.com/guidone/node-red-contrib-netatmo-dashboard

Already switched to that plugin - works without a flaw, all values (min, max, wind, rain) and no issues with refreshing token so far 🙏👍

**Comment by lost-carrier** [comment]
+1

...I guess this one finally kicked in:  https://github.com/ssadams11/node-red-contrib-netatmo/issues/27 :-(

**Comment by hflatoey** [comment]
> You can try this plugin for Netatmo: https://github.com/guidone/node-red-contrib-netatmo-dashboard

Thanks, it works :)

**Comment by lost-carrier** [comment]
> > You can try this plugin for Netatmo: https://github.com/guidone/node-red-contrib-netatmo-dashboard
> 
> Thanks, it works :)

Yes, authentication works with that one - however: apparently it can only fetch current data - no history…

**Comment by S474N** [comment]
For history data you can use this script:
https://github.com/michaelmiklis/netatmo.sh/issues/4#issuecomment-1636814038

**Comment by lost-carrier** [comment]
Thanks @S474N , but...

...I have a Java App (using https://github.com/lost-carrier/netatmo-api) which saves the last data received and retrieves everything newer. That proved to work very stable in grabbing the whole timeseries even if something boke down for some time.

Unfortunately that Java things now also stopped working due to Netatmo auth change. I wanted to use this opportunity to rebulid that process in NodeRed since that Java thingy tends to be a bit cumbersome to look into when only changing something every couple of years. But it looks like I will need dig into that thingy once again...

**Comment by stefan242** [comment]
> > > You can try this plugin for Netatmo: https://github.com/guidone/node-red-contrib-netatmo-dashboard
> > 
> > 
> > Thanks, it works :)
> 
> Yes, authentication works with that one - however: apparently it can only fetch current data - no history…

... and there you cannot request data from public Netatmo stations like with the node "get public data".
I used this to get rain data from other Netatmo installations in my area.
So I'm still waiting for an update which supports the new auth system with the refresh Token.

**Comment by mhin76** [comment]
same problem here. Dashboard nodes work, but as mentioned above, they do not provide historical data retrieval. A pity that the new authentication has not been implemented to this node here yet. Maybe the code could be ported from dashboard node to this one here?

**Comment by fjulian79** [comment]
Same issue

**Comment by mister-gaga** [comment]
Here also

**Comment by mdhom** [comment]
same here
